public/js/libs/helper.js

```diff
@@ -0,0 +1,129 @@
+const plotSingle = (graph, layer, transition = true) => {
+    graph.class_keep = graph.clickedLegend;
+    graph.idx = graph.legendIdentityArray.indexOf(graph.class_keep);
+
+    // Erase all but selected bars by setting opacity to 0
+    for (let i = 0; i < graph.legendIdentityArray.length; i++) {
+        if (graph.legendIdentityArray[i] != graph.class_keep) {
+            if (!transition) {
+                d3.selectAll(`.${graph.legendIdentityArray[i]}`).style("opacity", 0);
+            } else {
+                d3.selectAll(`.${graph.legendIdentityArray[i]}`).transition().duration(500).style("opacity", 0);
+            }
+        }
+    }
+
+    // Lower the bars to start on x-axis
+    graph.y_orig = []; // to store original y-posn
+    layer.selectAll("rect")._groups[graph.idx].forEach(function (d, i, n) {
+        // Get height and y posn of base bar and selected bar
+        let h_keep = d3.select(d).attr("height");
+        let y_keep = d3.select(d).attr("y");
+        // Store y_base in array to restore plot
+        graph.y_orig.push(y_keep);
+
+        let h_base, y_base;
+        layer.selectAll("rect")._groups[0].forEach(function (d, i, n) {
+            h_base = d3.select(d).attr("height");
+            y_base = d3.select(d).attr("y");
+        });
+
+        let h_shift = h_keep - h_base;
+        let y_new = y_base - h_shift;
+
+        // Reposition selected bars
+        if (!transition) {
+            d3.select(d).attr("y", y_new);
+        } else {
+            d3.select(d).transition().ease(d3.easeBounce).duration(500).delay(150).attr("y", y_new);
+        }
+    });
+    return graph;
+};
+
+const restorePlot = (graph, layer) => {
+    layer.selectAll("rect")._groups[graph.idx].forEach(function (d, i, n) {
+        d3.select(d).transition().duration(500).attr("y", graph.y_orig[i]);
+    });
+
+    //restore opacity of erased bars
+    for (let i = 0; i < graph.legendIdentityArray.length; i++) {
+        if (graph.legendIdentityArray[i] != graph.class_keep) {
+            d3.selectAll(`.${graph.legendIdentityArray[i]}`).transition().duration(500).delay(150).style("opacity", 1);
+        }
+    }
+};
+
+const cellOverHandler = (target, graph) => {
+    let legendHovered = d3.select(target).datum().replace(/\s/g, "");
+    if (graph.activeLink === "0") d3.select(target).style("cursor", "pointer");
+    else {
+        if (graph.activeLink === legendHovered) {
+            d3.select(target).style("cursor", "pointer");
+        } else d3.select(target).style("cursor", "auto");
+    }
+};
+
+const cellClickHandler = (target, graph, layer) => {
```

Comment:
Can you rename these handlers in terms of what they actually do rather than where they should be called?